ZelBack/src/services/appsService.js

```diff
@@ -8933,92 +8940,89 @@ async function checkMyAppsAvailability() {
-            log.warn(JSON.stringify(data));
-            currentDos += 0.4;
-            dosState += 0.4;
-          } else if (resMyAppAvailability && resMyAppAvailability.data.status === 'success') {
-            log.info(`${resMyAppAvailability.data.data.message} Detected from ${askingIP}:${askingIpPort}`);
-          }
-        } else {
-          log.info(`Running application ${app.name} on ports ${JSON.stringify(appPorts)} open but app not listening at them.`);
-        }
-      } else if (myAppAvailability && myAppAvailability.data.status === 'error') {
-        log.warn(`Running application ${app.name} on ports ${JSON.stringify(appPorts)}, unavailability detected from inside`);
-        log.warn(JSON.stringify(data));
-        currentDos += 0.4;
-        dosState += 0.4;
-      }
+    }
+    // choose random port
+    const min = config.fluxapps.portMin - 1000;
+    const max = config.fluxapps.portMax;
+    testingPort = Math.floor(Math.random() * (max - min) + min);
+    if (appPorts.includes(testingPort)) {
+      // skip this check
+      await serviceHelper.delay(60 * 60 * 1000);
+      checkMyAppsAvailability();
+      return;
+    }
+    // now open this port properly and launch listening on it
+    await fluxNetworkHelper.allowPort(testingPort);
+    if ((userconfig.initial.apiport && userconfig.initial.apiport !== config.server.apiport) || isUPNP) {
+      await upnpService.mapUpnpPort(testingPort, 'Flux_Test_App');
+    }
+    await serviceHelper.delay(5 * 60 * 1000);
+    testingAppserver.listen(testingPort);
+    await serviceHelper.delay(10 * 60 * 1000);
+    // eslint-disable-next-line no-await-in-loop
+    let askingIP = await fluxNetworkHelper.getRandomConnection();
+    let askingIpPort = config.server.apiport;
+    if (askingIP.includes(':')) { // has port specification
+      // it has port specification
+      const splittedIP = askingIP.split(':');
+      askingIP = splittedIP[0];
+      askingIpPort = splittedIP[1];
+    }
+    const timeout = 30000;
+    const axiosConfig = {
+      timeout,
+    };
+    const data = {
+      ip: myIP,
+      port: myPort,
+      appname: 'appPortsTest',
+      ports: [testingPort],
+      pubKey,
+    };
+    const stringData = JSON.stringify(data);
+    // eslint-disable-next-line no-await-in-loop
+    const signature = await signCheckAppData(stringData);
+    data.signature = signature;
+    // first check against our IP address
+    // eslint-disable-next-line no-await-in-loop
+    const resMyAppAvailability = await axios.post(`http://${askingIP}:${askingIpPort}/flux/checkappavailability`, JSON.stringify(data), axiosConfig).catch((error) => {
+      log.error(`${askingIP} for app availability is not reachable`);
+      log.error(error);
+    });
+    if (resMyAppAvailability && resMyAppAvailability.data.status === 'error') {
+      log.warn(`Applications port range unavailability detected from ${askingIP}:${askingIpPort}`);
```

Comment:
We should have here a while in error, or next time we execute the method it will be on a different random port that can be ok and we clear the DOS when in fact there are ports that aren't ok. And also because the method itself takes 15m delay on service listening creation.

Reply:
The chance of hitting the same testing port is 1 to 10000 basically. So it highly improbable of testing the same port. We are also not testing ports that are in use per apps. I think this is sufficient and if lets say 1000 ports are not working well in the range (aka node op is artificially blocking some port range), it is not ideal yes - majority of apps will run well but some not.
Now its a question if node op is blocking one port - that is 'fine' but if node op is blocking 50% of ports, that is not fine.
So some adjustment to this test would be in fact needed. As if 50% of ports are blocked, our test will still allow it and said that this flux is ok

Reply:
resolved. If port fails, we will be testing this port again if unused.